zcgo/MakeDefinesCommand.php

```diff
@@ -0,0 +1,299 @@
+ * @version $Id:  $
+ */
+
+namespace Zencart\Go;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputDefinition;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Zencart\Go\exceptions\InputOptionsValidationException;
+use Zencart\FileSystem\FileSystem;
+
+class MakeDefinesCommand extends Command
+{
+    protected static $defaultName = 'make:defines';
+
+    protected function configure()
+    {
+        $this
+            ->setDescription('Creates an array based language define file.')
+            ->setHelp(
+                'This command allows you to create an array based language file based on a legacy define file.' . "\n" .
+                ' You can use the options to either select a file or directory to convert, or use a config file for more control. ')
+            ->setDefinition(
+                new InputDefinition(
+                    [
+                        new InputOption('file', 'f', InputOption::VALUE_REQUIRED),
+                        new InputOption('dir', 'd', InputOption::VALUE_REQUIRED),
+                        new InputOption('config', 'c', InputOption::VALUE_REQUIRED),
+                    ])
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->input = $input;
+        $this->output = $output;
+        $this->fileSystem = FileSystem::getInstance();
+        try {
+            $this->validateInputOptions($input);
+        } catch (InputOptionsValidationException $e) {
+            $output->writeln($e->getMessage());
+            exit(1);
+        }
+        try {
+            $this->processFilesFromInput($input);
+        } catch (InputOptionsValidationException $e) {
+            $output->writeln($e->getMessage());
+            exit(1);
+        }
+        return 0;
+    }
+
+    protected function processFilesFromInput($input)
+    {
+        $file = $input->getOption('file');
+        $dir = $input->getOption('dir');
+        $config = $input->getOption('config');
+        try {
+            if (isset($file)) {
+                $this->processSingleFile($file);
+            }
+            if (isset($dir)) {
+                $this->processDirectory($dir);
+            }
+            if (isset($config)) {
+                $this->processConfigFile($config);
+            }
+        } catch (InputOptionsValidationException $e) {
+            throw new InputOptionsValidationException($e->getMessage());
+        }
+    }
+
+    protected function validateInputOptions($input)
+    {
+        $file = $input->getOption('file');
+        $dir = $input->getOption('dir');
+        $config = $input->getOption('config');
+        if (!isset($file) && !isset($dir) && !isset($config)) {
+            throw new InputOptionsValidationException('Seems you didn\'t pass any options');
```

Comment:
Shouldn't this read 'Seems you didn\'t pass all options'? Using `all` instead of `any`?